src/main/java/apoc/sequence/Sequence.java

```diff
@@ -0,0 +1,151 @@
+package apoc.sequence;
+
+import apoc.result.NodeResult;
+import apoc.result.ObjectResult;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.procedure.*;
+
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+/**
+ * Created by Valentino Maiorca on 6/29/17.
+ */
+public class Sequence {
+
+    /**
+     * Label for all sequence-node.
+     */
+    public static final Label SEQUENCE_LABEL = Label.label("Apoc_Sequence");
+
+    public static final String PROVIDER_PROPERTY = "provider";
+
+    public static final String NAME_PROPERTY = "name";
+
+    // Simple provider. Increments by 1 the value.
+    public static final SequenceProvider<Long> LONG_SP = new SequenceProvider<Long>() {
+
+        @Override
+        public Long start() {
+            return 0L;
+        }
+
+        @Override
+        public Long next(Long current) {
+            return current + 1;
+        }
+
+        @Override
+        public LinkedList<Long> next(Long current, long length) {
```

Comment:
why a linkedlist and not just an long-stream?
also please don't use implementation classes as return values